Packs/Tenable_io/Integrations/Tenable_io/Tenable_io.py

```diff
@@ -1114,33 +1116,269 @@ def export_vulnerabilities_command(args: Dict[str, Any]) -> PollResult:
+        'scan_uuid': 'History uuid',
+        'status': 'Status',
+        'is_archived': 'Is archived',
+        'custom': 'Targets custom',
+        'default': 'Targets default',
+        'visibility': 'Visibility',
+        'time_start': 'Time start',
+        'time_end': 'Time end',
+    }
+    return tableToMarkdown(
+        'Tenable IO Scan History',
+        list(map(flatten, history)),
+        headers=list(context_to_hr.keys()),
+        headerTransform=context_to_hr.get)
+
+
+def scan_history_params(args: dict) -> dict:
+    return {
+        'sort': ','.join(
+            f'{field}:{args["sortOrder"]}'
+            for field in argToList(args.get('sortFields'))),
+        'exclude_rollover': args['excludeRollover'],
+        'limit': args.get('pageSize') or args['limit'],
+        'offset': int(args.get('pageSize', 0)) * int(args.get('page', 0)),
+    }
+
+
+def get_scan_history_command(args: dict[str, Any]) -> CommandResults:
+
+    response_dict = get_scan_history_request(
+        args['scanId'],
+        scan_history_params(args))
+    history = response_dict.get('history', [])
+
+    return CommandResults(
+        outputs_prefix='TenableIO.ScanHistory',
+        outputs_key_field='id',
+        outputs=history,
+        readable_output=scan_history_human_readable(history),
+        raw_response=response_dict)
+
+
+def export_scan_request_args(
+        scanId: str, historyId=None, historyUuid=None, chapters=None,
+        filterSearchType=None, assetId=None, filterName=None,
+        filterQuality=None, filterValue=None, **args) -> dict:
+
+    if chapters:
+        chapters = ';'.join(argToList(chapters))
+    elif args['format'] in ('PDF', 'HTML'):
+        raise DemistoException('The "chapters" field must be provided for PDF or HTML formats.')
+
+    url = f'{BASE_URL}scans/{scanId}/export'
+    params = {
+        'history_id': historyId,
+        'history_uuid': historyUuid,
+    }
+    body = {
+        'format': args['format'].lower(),
+        'chapters': chapters,
+        'filter.search_type': filterSearchType,
+        'asset_id': assetId,
+    }
+    for i, (name, quality, value)\
+            in enumerate(zip(*map(argToList, (filterName, filterQuality, filterValue)))):
+        body |= {
+            f'filter.{i}.filter': name,
+            f'filter.{i}.quality': quality,
+            f'filter.{i}.value': value,
+        }
+    remove_nulls_from_dictionary(params)
+    remove_nulls_from_dictionary(body)
+    return {'url': url, 'params': params, 'json': body}
+
+
+def initiate_export_scan(scan_id: str, args: dict) -> PollResult:
+    response = requests.post(
+        **export_scan_request_args(**args),
+        headers=HEADERS, verify=USE_SSL)
+    response_json = safe_get_json(response)
+
+    if not (file_id := response_json.get('file')):
+        raise DemistoException(
+            f'Unexpected response from Tenable IO: {response_json}')
+
+    return PollResult(
+        None, True,
```

Comment:
can you label these params? Did we finish here? Why are they needed?

Reply:
labeled